Source/WebKit/UIProcess/mac/WebContextMenuProxyMac.mm

```diff
@@ -512,6 +512,27 @@ static void getStandardShareMenuItem(NSArray *items, void (^completionHandler)(N
 
         return true;
     }
+
+    if (auto potentialQRCodeNodeSnapshotImage = m_context.potentialQRCodeNodeSnapshotImage()) {
+        auto image = potentialQRCodeNodeSnapshotImage->makeCGImage();
+        requestPayloadForQRCode(image.get(), [this, protectedThis = Ref { *this }](NSString *result) mutable {
+            if (!result.length) {
```

Comment:
can this maybe be extracted into it's own method? the chain of callbacks / if statements here is confusing

Reply:
Will try to clean it up a bit.

Reply:
I felt it made sense to keep this in the same method, but I have re-written the logic with readability in mind, and reduced overall indentation by two levels.